frontend/apps/pablo/components/Organisms/AllBondTable.tsx

```diff
@@ -39,12 +44,110 @@ const tableHeaders: TableHeader[] = [
 
 const BOND_LIMIT_TO_SHOW = 4;
 
+const BondPrincipalAssetIcon = ({
+  principalAsset,
+}: {
+  principalAsset: BondPrincipalAsset;
+}) => {
+  const { lpPrincipalAsset, simplePrincipalAsset } = principalAsset;
+  const { baseAsset, quoteAsset } = lpPrincipalAsset;
+
+  if (baseAsset && quoteAsset) {
+    return (
+      <PairAsset
+        assets={[
+          {
+            icon: baseAsset.icon,
+            label: baseAsset.symbol,
+          },
+          {
+            icon: quoteAsset.icon,
+            label: quoteAsset.symbol,
+          },
+        ]}
+        separator="/"
+      />
+    );
+  }
+
+  if (simplePrincipalAsset) {
+    return (
+      <BaseAsset
+        label={simplePrincipalAsset.symbol}
+        icon={simplePrincipalAsset.icon}
+      />
+    );
+  }
+
+  return null;
+};
+
+const BondOfferRow = ({
+  bondOffer,
+  handleBondClick,
+}: {
+  bondOffer: BondOffer;
+  handleBondClick: (bondOfferId: string) => void;
+}) => {
+  const roi = useBondOfferROI(bondOffer);
+  const totalPurchasedValue = useTotalPurchasedBondOffer(bondOffer);
+  const principalAsset = useBondOfferPrincipalAsset(bondOffer);
+  const bondPrice = useBondPrice(bondOffer);
+
+  return (
+    <TableRow
+      key={bondOffer.offerId.toString()}
+      onClick={() => handleBondClick(bondOffer.offerId.toString())}
+      sx={{ cursor: "pointer" }}
+    >
+      <TableCell align="left">
+        <BondPrincipalAssetIcon principalAsset={principalAsset} />
+      </TableCell>
+      <TableCell align="left">
+        <Typography variant="body2">${bondPrice.toFormat()}</Typography>
+      </TableCell>
+      <TableCell align="left">
+        <Typography variant="body2" color="featured.main">
+          {roi.toFormat()}%
+        </Typography>
+      </TableCell>
+      <TableCell align="left">
+        <Typography variant="body2">
+          ${totalPurchasedValue.toFormat()}
+        </Typography>
+      </TableCell>
+    </TableRow>
+  );
+};
+
 export const AllBondTable: React.FC = () => {
   const theme = useTheme();
   const router = useRouter();
-  const bondOffers = useBondOffers();
-  const [count, setCount] = useState(BOND_LIMIT_TO_SHOW);
+  const { parachainApi } = useParachainApi(DEFAULT_NETWORK_ID);
+
+  const {
+    bondOffers: { list },
+    setBondOffers,
+    setBondOfferTotalPurchased,
+  } = useStore();
+
+  useEffect(() => {
+    if (parachainApi) {
+      fetchBondOffers(parachainApi).then((decodedOffers) => {
+        setBondOffers(decodedOffers);
+      });
+    }
+  }, [parachainApi, setBondOffers]);
 
+  useEffect(() => {
+    fetchTotalPurchasedBondsByOfferIds().then(
+      (totalPurchasedByOfferIds) => {
+        setBondOfferTotalPurchased(totalPurchasedByOfferIds);
+      }
+    );
```

Comment:
```suggestion
    fetchTotalPurchasedBondsByOfferIds().then(
      setBondOfferTotalPurchased
    );
```

We can go point-free style here too.